src/PointwiseFunctions/AnalyticSolutions/EinsteinSolutions/KerrSchild.cpp

```diff
@@ -0,0 +1,172 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/AnalyticSolutions/EinsteinSolutions/KerrSchild.hpp"
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/Magnitude.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace EinsteinSolutions {
+
+KerrSchild::KerrSchild(const double mass, KerrSchild::Spin::type spin,
+                       KerrSchild::Center::type center,
+                       const OptionContext& /* context */) noexcept
+    : mass_(mass),
+      // clang-tidy: do not std::move trivial types.
+      center_(std::move(center)),  // NOLINT
+      // clang-tidy: do not std::move trivial types.
+      spin_(std::move(spin))  // NOLINT
+{
+  const double spin_magnitude = magnitude(spin_);
+  if (spin_magnitude > 1.0) {
+    ERROR("Spin magnitude must be < 1. Given spin: "
+          << spin_ << " with magnitude " << spin_magnitude);
+  }
+  if (mass_ < 0.0) {
+    ERROR("Mass must be non-negative. Given mass: " << mass_);
+  }
+}
+
+Variables<KerrSchild::Tags> KerrSchild::solution(
+    const tnsr::I<DataVector, 3>& x, const double /*t*/) const noexcept {
```

Comment:
This function has a lot of temporaries which can be made const by initialising using a lambda.